src/kindlemint/agents/kdp_performance_agent.py

```
@@ -0,0 +1,652 @@
+
+    async def _scrape_amazon_product_page(self, asin: str) -> Dict[str, Any]:
+        """Scrape public Amazon product page data"""
+        headers = {
+            "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36"
+        }
+
+        url = f"https://www.amazon.com/dp/{asin}"
+
+        try:
+            async with aiohttp.ClientSession(headers=headers) as session:
+                async with session.get(url) as response:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)